### PR TITLE
refactor: Introduce `RatchetTreeExported` to unify `Vec<Option<Node>>`-handling.

### PR DESCRIPTION
**Pull request by duesee**
This PR introduces a `RatchetTreeExported` type. This is useful for (later) validation and improves readability. This PR also changes the pattern of "borrowing and cloning internally" to "moving" (and letting the caller decide when to clone.)

### CODE REVIEW

**Comment by franziskuskiefer** [comment]
Note that the spec changes here to make this stricter in mlswg/mls-protocol#868

File: openmls/src/treesync/mod.rs [hunk 1]
```diff
@@ -68,6 +71,102 @@ pub use node::{leaf_node::LeafNode, parent_node::ParentNode, Node};
...
+    Clone,
+    Debug,
+    Default,
+    Serialize,
+    Deserialize,
+    TlsSerialize,
+    TlsDeserialize,
+    TlsSize,
+)]
+// TODO(validation):
```
Comment (duesee) [inline openmls/src/treesync/mod.rs:87]: We could already open an issue for that.

Comment (raphaelrobert) [reply openmls/src/treesync/mod.rs:94]: Yes, please convert this into TODOs with a ticket number

File: openmls/src/treesync/mod.rs [hunk 2]
```diff
@@ -68,6 +78,102 @@ pub use node::{leaf_node::LeafNode, parent_node::ParentNode, Node};
...
+//
+// * ValSemXXX: "If the tree has 2^d leaves, then it has 2^(d+1) - 1 nodes. The ratchet_tree vector logically has this number of entries, but the sender SHOULD NOT include blank nodes after the last non-blank node."
+// * ValSemXXX: "Regardless of how the client obtains the tree, the client MUST verify that the root hash of the ratchet tree matches the tree_hash of the GroupContext before using the tree for MLS operations."
+pub struct RatchetTreeExported(Vec<Option<Node>>);
```
Comment (raphaelrobert) [inline openmls/src/treesync/mod.rs:98]: Why not generically call this `RatchetTree`? That way we can have a `RatchetTreeIn` in the context of validation.

Comment (duesee) [reply openmls/src/treesync/mod.rs:98]: No good reason. Happy to rename it to `RatchetTree` when you don't see potential for confusion :+1:

File: openmls/src/treesync/mod.rs (hunk 1 above)
Comment (duesee) [reply openmls/src/treesync/mod.rs]: I created #1305.

File: openmls/src/treesync/mod.rs (hunk 2 above)
Comment (duesee) [reply openmls/src/treesync/mod.rs]: Ah... I renamed it to `RatchetTree` but could have already used `RatchetTreeIn`. Should we leave it that way anticipating this change or rename later? I'm leaning towards renaming later.

**Review by raphaelrobert — APPROVED** [review]

**Comment by raphaelrobert** [comment]
`cargo doc` is unhappy, otherwise we are good to go

**Review by raphaelrobert — APPROVED** [review]

**Pull request merged**